credentials/tls/certprovider/pemfile/builder.go

```diff
@@ -61,7 +60,7 @@ func (p *pluginBuilder) ParseConfig(c interface{}) (*certprovider.BuildableConfi
 		if err != nil {
 			// This should never really happen since we validate the config in
 			// ParseConfig and we do not use the BuildOptions passed in here.
-			logger.Errorf("NewProvider(%+v) failed: %v", err)
+			logger.Error(err)
```

Comment:
Ohh.. I mis-read this as a `return` -- I forgot this builder doesn't have an error.  In that case what you had before was totally fine.

Although what happens if we return a `nil` provider?  Will we get a panic?  If so, `log.Fatal` would be better here.

OR, can we have a non-exported constructor that can't error?

Reply:
I reverted to what I had earlier.

With regards to what happens when we return a `nil` provider is we don't panic:
https://github.com/grpc/grpc-go/blob/20636e76a99aed65ef8706949d2400079ba003da/credentials/tls/certprovider/store.go#L119

Having a non-exported constructor makes sense. Added it.